Make configmaps mutable unless marked explicitly (#1926)
pulumi/pulumi-kubernetes: CHANGELOG.md

## Unreleased

> -  Make configmaps mutable unless marked explicitly (enabled with provider config option) (https://github.com/pulumi/pulumi-kubernetes/pull/1926)

Reviewer: ```suggestion
-  Make ConfigMaps mutable unless marked explicitly (enabled with provider config option) (https://github.com/pulumi/pulumi-kubernetes/pull/1926)
```

>    *NOTE*: With this change, once `enableConfigMapMutable` is enabled, all config maps will be seen as mutable. The replacement behavior can be opted in by adding `replaceOnChanges: ["*"]` etc.

Reviewer: ```suggestion
   *NOTE*: With this change, once `enableConfigMapMutable` is enabled, all ConfigMaps will be seen as mutable. In this mode, you can opt into in the previous replacement behavior for a particular ConfigMap by setting its `replaceOnChanges` resource option to `[".binaryData", ".data"]`.
```

>    Without the provider option enabled, the existing semantics of treating config maps as immutable will hold. i.e. mutations of configMaps will trigger a replacement.

Reviewer: ```suggestion
   By default, the provider will continue to treat ConfigMaps as immutable, and will replace them if the `binaryData` or `data` properties are changed.
```
